HCCGo/app/js/jobSubmissionCtrl.js

```diff
@@ -124,21 +143,43 @@ jobSubmissionModule.controller('jobSubmissionCtrl', ['$scope', '$log', '$timeout
 
     $("#submitbtn").prop('disabled', true);
 
+    // Separate SBATCH options from commands
+    job.commands = editor.getValue();
+    var other = editor.getValue().split("\n");
+    var sbatch = [];
+    for(var x = 0; x < other.length; x++) {
+      if (other[x].startsWith("#SBATCH")) {
+        sbatch.push(other[x]);
+        other.splice(x,1);
```

Comment:
Since you're splicing, you can't iterate over an array without doing something with x.  IE, you are removing an element, so everything shifts down 1 place, but X is still increasing.  So you are missing lines.